proc/proc_windows.go

```diff
@@ -126,13 +121,56 @@ func Launch(cmd []string) (*Process, error) {
 		dbp.postExit()
 		return nil, ProcessExitedError{Pid: dbp.Pid, Status: exitCode}
 	}
+	return initializeDebugProcess(dbp, exepath, false)
+}
+
+// findExePath searches for process pid, and returns its executable path.
+func findExePath(pid int) (string, error) {
+	// Original code suggested different approach (see below).
+	// Maybe it could be useful in the future.
+	//
+	// Find executable path from PID/handle on Windows:
+	// https://msdn.microsoft.com/en-us/library/aa366789(VS.85).aspx
+	//
+
+	p, err := syscall.OpenProcess(syscall.PROCESS_QUERY_INFORMATION, false, uint32(pid))
+	if err != nil {
+		return "", err
+	}
+	defer syscall.CloseHandle(p)
 
-	return initializeDebugProcess(dbp, argv0Go, false)
+	n := uint32(128)
+	for {
+		buf := make([]uint16, int(n))
+		err = _QueryFullProcessImageName(p, 0, &buf[0], &n)
+		switch err {
+		case syscall.ERROR_INSUFFICIENT_BUFFER:
+			// try bigger buffer
+			n *= 2
+			// but stop if it gets too big
+			if n > 10000 {
+				return "", err
+			}
+		case nil:
+			return syscall.UTF16ToString(buf[:n]), nil
+		default:
+			return "", err
+		}
+	}
 }
 
 // Attach to an existing process with the given PID.
 func Attach(pid int) (*Process, error) {
-	return nil, fmt.Errorf("not implemented: Attach")
+	// TODO: Probably should have SeDebugPrivilege before starting here.
```

Comment:
I am not clear what you are asking about. If you asking if we need to hold SeDebugPrivilege when we are starting debugged process ourselves (as well as when attaching to external process), then I do not know. CreateProcess Windows API does not mention SeDebugPrivilege, while DebugActiveProcess does.

I played around with new "attach" functionality, and I can attach to external process (small Go program) with code in this PR. So it seems I don't always need to ask for SeDebugPrivilege. But I have no doubt we will find ourselves in a situation when this is required. So I have added a TODO, so we don't forget. I suspect we could try acquire SeDebugPrivilege once at the start of the program, and continue with our job regardless of result - maybe display warning message if SeDebugPrivilege could not be acquired.